t/t3907-stash-show-config.sh

```diff
@@ -0,0 +1,94 @@
+#!/bin/sh
+
+test_description='Test git stash show configuration.
+
+'
+
+. ./test-lib.sh
+
+config_stash_stat_and_patch () {
+	if test "$1" != "unset"
+	then
+		git config stash.showStat "$1"
```

Comment:
If you use `test_config`, then the config setting will be automatically reverted after the test case. Not sure whether you want to do that here, but I could imagine that you do....